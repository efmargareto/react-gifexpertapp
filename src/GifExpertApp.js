import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [ categorias, setCategorias ] = useState( defaultCategories ) // []

    return (
        <>
            <h2>GifExpertApp</h2>

            <AddCategory setCategorias={ setCategorias }/>

            <hr />
            <ol>
                {
                    categorias.map( category => (

                        <GifGrid 
                            key={ category }    
                            category={ category }
                        />
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpertApp
