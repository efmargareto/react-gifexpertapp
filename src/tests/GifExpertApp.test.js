import { shallow } from 'enzyme'
import React from 'react'
import GifExpertApp from '../GifExpertApp'


describe('Pruebas componente <GifExpertApp />', () => {

    test('Debe concordar con el smashShot', () => {
        
        const wrapper = shallow( <GifExpertApp /> )

        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe de mostrar un lista de categorias', () => {

        const categories = ['Wow', 'divertido']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> )

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    })
    
        
})
