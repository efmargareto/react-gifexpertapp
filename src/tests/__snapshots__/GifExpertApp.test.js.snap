// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifExpertApp /> Debe concordar con el smashShot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas componente <GifExpertApp /> Debe de mostrar un lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Wow"
      key="Wow"
    />
    <GifGrid
      category="divertido"
      key="divertido"
    />
  </ol>
</Fragment>
`;
